test/junit/src/test/scala/strawman/collection/ArrayOpsTest.scala

```diff
@@ -0,0 +1,71 @@
+package strawman.collection
+
+import org.junit.Assert._
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import strawman.collection.immutable.List
+
+@RunWith(classOf[JUnit4])
+class ArrayOpsTest {
+
+  @Test
+  def hasCorrectOverloadedZipWhenInputHasSameNumberOfElements: Unit = {
+    val array: ArrayOps[Int] = Array(1, 2, 3)
+    val result: Array[(Int, String)] = array.zip(List("a", "b", "c"))
+
+    assertArrayEquals(Array((1, "a"), (2, "b"), (3, "c")).asInstanceOf[Array[AnyRef]], result.asInstanceOf[Array[AnyRef]])
```

Comment:
Isn’t it working if you omit the `asInstanceOf`s?

Reply:
no, it does not compile:
```
overloaded method value assertArrayEquals with alternatives:
[error]   (x$1: Array[Long],x$2: Array[Long])Unit <and>
[error]   (x$1: Array[Int],x$2: Array[Int])Unit <and>
[error]   (x$1: Array[Short],x$2: Array[Short])Unit <and>
[error]   (x$1: Array[Char],x$2: Array[Char])Unit <and>
[error]   (x$1: Array[Byte],x$2: Array[Byte])Unit <and>
[error]   (x$1: Array[Object],x$2: Array[Object])Unit
[error]  cannot be applied to (Array[(Int, String)], Array[(Int, String)])
[error]     assertArrayEquals(Array((1, "a"), (2, "b"), (3, "c")), result)
[error]     ^
```

Reply:
What if you replace that with `assertTrue(Array((1, "a"), ...).equals(result))`?

Reply:
`equals` is not overwritten so it does not compare the content of `Array` objects. But I can do what you suggested with `sameElements` so that these asserts become shorter.